Remotion/ObjectBinding/Web.UnitTests/UI/Controls/Validation/ReadOnlyBusinessObjectValidationResultDecoratorTest.cs

```diff
@@ -0,0 +1,61 @@
+﻿// This file is part of the re-motion Core Framework (www.re-motion.org)
+// Copyright (c) rubicon IT GmbH, www.rubicon.eu
+//
+// The re-motion Core Framework is free software; you can redistribute it
+// and/or modify it under the terms of the GNU Lesser General Public License
+// as published by the Free Software Foundation; either version 2.1 of the
+// License, or (at your option) any later version.
+//
+// re-motion is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public License
+// along with re-motion; if not, see http://www.gnu.org/licenses.
+//
+using System;
+using Moq;
+using NUnit.Framework;
+using Remotion.ObjectBinding.Validation;
+using Remotion.ObjectBinding.Web.UI.Controls.Validation;
+
+namespace Remotion.ObjectBinding.Web.UnitTests.UI.Controls.Validation
+{
+  [TestFixture]
+  public class ReadOnlyBusinessObjectValidationResultDecoratorTest
+  {
+    [Test]
+    public void GetValidationFailures_AlwaysPassesMarkAsHandledFalse ()
+    {
+      var businessObject = Mock.Of<IBusinessObject>();
+      var businessObjectProperty = Mock.Of<IBusinessObjectProperty>();
+
+      var validationResultMock = new Mock<IBusinessObjectValidationResult>(MockBehavior.Strict);
+      validationResultMock.Setup(e => e.GetValidationFailures(businessObject, businessObjectProperty, false))
+          .Returns(Array.Empty<BusinessObjectValidationFailure>())
+          .Verifiable();
+
+      var readOnlyDecorator = new ReadOnlyBusinessObjectValidationResultDecorator(validationResultMock.Object);
+      readOnlyDecorator.GetValidationFailures(businessObject, businessObjectProperty, true);
```

Comment:
I think we should test that the inner result is passed to the callsite (return collection instance and use `Is.SameAs`). Otherwise, we could return anything from `null` to `Array.Emtpy` and no one would know.